pkg/restore/restic_restore_action.go

```diff
@@ -71,10 +75,23 @@ func (a *ResticRestoreAction) Execute(input *velero.RestoreItemActionExecuteInpu
 
 	log := a.logger.WithField("pod", kube.NamespaceAndName(&pod))
 
-	volumeSnapshots := restic.GetPodSnapshotAnnotations(&pod)
+	opts := restic.NewPodVolumeBackupListOptions(input.Restore.Spec.BackupName)
+	podVolumeBackupList, err := a.podVolumeBackupClient.List(opts)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	var podVolumeBackups []*velerov1api.PodVolumeBackup
+	for i, pvb := range podVolumeBackupList.Items {
+		podVolumeBackups[i] = &pvb
+	}
+	volumeSnapshots := restic.GetVolumesForPod(podVolumeBackups, &pod)
 	if len(volumeSnapshots) == 0 {
-		log.Debug("No restic snapshot ID annotations found")
-		return velero.NewRestoreItemActionExecuteOutput(input.Item), nil
+		volumeSnapshots = restic.GetPodSnapshotAnnotations(&pod)
```

Comment:
it looks like you could change the `restic.GetVolumesForPod` helper to first check the PVBs, then check the annotations if there are no associated PVBs - that way you don't have to repeat this code in a couple different places. I think that makes more sense.